content/spin/rust-components.md

```diff
@@ -367,6 +367,61 @@ messages on the `messages` Redis channel.
 
 Spin has a key-value store built in. For information about using it from Rust, see [the key-value store tutorial](kv-store).
 
+If you want to go beyond just inserting values in a certain spot, Spin's Rust SDK provides an optional feature for using [Serde](https://docs.rs/serde/latest/serde). The Rust code below shows how to store and retrieve serializable objects from the key-value store:
+
+```rust
+use anyhow::Result;
+use spin_sdk::{
+    http::{Request, Response},
+    http_component,
+    key_value::{Store},
+};
+use serde::{Serialize, Deserialize};
+
+#[http_component]
+fn handle_request(_req: Request) -> Result<Response> {
+    // Open the default key-value store
+    let store = Store::open_default()?;
+    // Define a User object using serde derive
+    #[derive(Serialize, Deserialize, Debug)]
+    struct User {
+        fingerprint: String,
+        location: String,
+    }
+    // Create an instance of a user object and populate the values
+    let user = User{fingerprint:"0x1234".to_owned(), location:"Brisbane".to_owned()};
+    // Store the user object using the "my_json" key
+    store.set_json(String::from("my_json"), &user)?;
+    // Retrieve the user object from the key-value store, using the "my_json" key
+    let retrieved_user: User = store.get_json("my_json")?;
+    // Respond to the request by revealing the user's fingerprint as the response body
+    Ok(http::Response::builder().status(200).body(Some(retrieved_user.fingerprint.into()))?)
+}
+```
+
+The above example requires some additional configuration. Specifically, enabling the Spin SDK's optional `json` feature (from v1.2.0 onwards) and adding Serde's optional `derive` feature in the `dependencies` section of the application's `Cargo.toml` file:
```

Comment:
The Rust templates don't reference `serde` by default so suggest rephrasing the second part, but I'm not sure if this is too wordy:

```suggestion
To use the `get_json` and `set_json` helpers, you must enable the Spin SDK's optional `json` feature in your `Cargo.toml` file.  To make your objects serialisable, you will also need a reference to `serde`; the example above also requires Serde's `derive` feature.  The relevant `Cargo.toml` entries look like this:
```

I wonder if it's worth adding a "details" expander along the lines of "Why are the helpers under an optional feature? // `get_json` and `set_json` depend on the `serde` and `serde_json` libraries. These are large libraries, so applications that depend on them are larger and take longer to build.  We didn't want users of the Rust SDK to incur these costs unless they actually wanted the feature."  Not ideal wording I fear but hopefully you get the idea!